src/Compilers/Core/Portable/Syntax/ChildSyntaxList.cs

```diff
@@ -83,17 +83,26 @@ private static int Occupancy(GreenNode green)
             return green.IsList ? green.SlotCount : 1;
         }
 
+        internal record struct SlotData(int SlotIndex, int FirstIndexAtSlotIndex, int PositionAtSlotIndex);
```

Comment:
do we need this to be a record?  is that basically just to expose properties?  you might want to consider making this a normal struct, with primary constructor and fields/properties.  Otherwise, we're getting all the equals/hashcode/operators that we aren't using.

Reply:
Changed to be a standard record with a primary ctor.